retrieval/transform.go

```diff
@@ -120,6 +121,23 @@ func (b *sampleBuilder) next(ctx context.Context, samples []tsdb.RefSample) (*mo
 	return &ts, entry.hash, samples[1:], nil
 }
 
+func (b *sampleBuilder) seriesGetWithRetry(ctx context.Context, sample tsdb.RefSample) (*seriesCacheEntry, bool) {
+	backoff := time.Duration(0)
+	entry, ok, err := b.series.get(ctx, sample.Ref)
+	for {
```

Comment:
This doesn't seem to honor ctx.Done anymore. Is that intended?

Reply:
now ctx.Done() is captured in this function. If context cancellation happens, it will at least be caught.